### PR TITLE
Add at-blasfunc

### PR DESCRIPTION
**Pull request by martinholters**
Trying to do some minimal subset of https://github.com/JuliaLang/julia/pull/16274 for a project, I found this to be useful.

### CODE REVIEW

**Comment by andreasnoack** [comment]
Both things sound great.

File: README.md [hunk 1]
```diff
@@ -209,6 +209,8 @@ Currently, the `@compat` macro supports the following syntaxes:
 
 * `@functorize` (not present in any Julia version) takes a function (or operator) and turns it into a functor object if one is available in the used Julia version. E.g. something like `mapreduce(Base.AbsFun(), Base.MulFun(), x)` can now be written as `mapreduce(@functorize(abs), @functorize(*), x)`, and `f(::Base.AbsFun())` as `f(::typeof(@functorize(abs)))`, to work across different Julia versions. `Func{1}` can be written as `supertype(typeof(@functorize(abs)))` (and so on for `Func{2}`), which will fall back to `Function` on Julia 0.5.
 
+* `Compat.@blasfunc` makes functionality of `Blas.@blasfunc` available on older Julia versions
```
Comment (TotalVerb) [inline README.md]: `Base.LinAlg.BLAS.@blasfunc`, not `Blas.@blasfunc`

File: src/Compat.jl [hunk 2]
```diff
@@ -1358,4 +1358,14 @@ else
     import Base.promote_eltype_op
 end
 
+if VERSION >= v"0.5.0-dev+4338"
+    import Base.LinAlg.BLAS.@blasfunc
+elseif  VERSION >= v"0.5.0-dev+1871"
```
Comment (TotalVerb) [inline src/Compat.jl]: extra space

File: test/runtests.jl [hunk 3]
```diff
@@ -1291,3 +1291,10 @@ end
 
 @test Compat.promote_eltype_op(@functorize(+), ones(2,2), 1) === Float64
 @test Compat.promote_eltype_op(@functorize(*), ones(Int, 2), zeros(Int16,2)) === Int
+
+let n=5, a=rand(n), incx=1, b=rand(n), incy=1
+    ccall((@Compat.blasfunc(dcopy_), Base.LinAlg.BLAS.libblas), Void,
```
Comment (TotalVerb) [inline test/runtests.jl]: minor style point: I think `Compat.@blasfunc` is more typical?

**Comment by martinholters** [comment]
Addressed @TotalVerb's comments. Also fixed some unrelated whitespace issues while at it (second commit).

**Comment by TotalVerb** [comment]
LGTM.

**Pull request merged**